### PR TITLE
Fix sequence inheritance defined in yaml

### PR DESCRIPTION
**Pull request by abcdefg30**
Fixes a regression from #20580.

Nodes need to be placed after "Inherits" nodes so that they take precedence over the inherited nodes.

Testcase: Build a harvester (or refinery) in Red Alert. On bleed the harvester will always have the "full" artwork.

### CODE REVIEW

**Comment by PunkPun** [comment]
I created an update rule to fix these, and it detected more errors.

https://github.com/OpenRA/OpenRA/compare/bleed...PunkPun:OpenRA:Fix-sequences

**Comment by abcdefg30** [comment]
> I created an update rule to fix these, and it detected more errors.

Thanks. However, I didn't spot any additional instances of that when comparing the rebased branches. But you did reveal that some of the new D2k maps have ill-formatted map.yaml files.

File: OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs [hunk 1]
```diff
@@ -315,8 +326,6 @@ public override IEnumerable<string> UpdateSequenceNode(ModData modData, MiniYaml
 			foreach (var sequenceNode in imageNode.Value.Nodes.ToList())
 				if (implicitInheritedSequences.Contains(sequenceNode.Key) && !sequenceNode.Value.Nodes.Any())
 					imageNode.RemoveNode(sequenceNode);
-
-			yield break;
```
Comment (PunkPun) [inline OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs]: Every update rule yields break. Why are you removing it?

Comment (abcdefg30) [reply OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs]: It's at the end of the method, therefore unnecessary:
image

Comment (PunkPun) [reply OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs]: Does it say that for all update rules?

Comment (abcdefg30) [reply OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs]: In fact it does, but I'm not sure if it's worth touching every update rule over it. I can revert this change if you think it is too out of place.

Comment (PunkPun) [reply OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs]: This rule doesn't exist in https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/

It seems this is something picked up by SonarSource. I'd rather you revert the change here, and we then do a follow up PR where we remove all redundant jumps

**Review by PunkPun — COMMENTED** [review]
Tested the update rule, it now generates defaults in the correct position

**Review by PunkPun — DISMISSED** [review]

File: OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs (hunk 1 above)
Comment (abcdefg30) [reply OpenRA.Mods.Common/UpdateRules/Rules/20221203/ExplicitSequenceFilenames.cs]: Reverted.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/013c7b221e9a87470f3ad5afaf48526c2e7dff45...8a339e0397e87ee02cbcc939a10a530ebcb8b4ca)

**Pull request merged**